SongController.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\StoreSong;
+use App\Song;
+use Illuminate\Http\Request;
+use Illuminate\Support\Str;
+use Goutte\Client;
+use Tymon\JWTAuth\Facades\JWTAuth;
+
+class SongController extends Controller
+{
+     /**
+     * Display a listing of songs.
+     *
+     * @return Response
+     */
+    public function index()
+    {
+        $songs = Song::get(['id', 'slug', 'artist', 'title'])->toArray();
+
+        return $songs;
+    }
+
+    /**
+     * Display a listing of songs that needs to be verified.
+     *
+     * @return Response
+     */
+    public function verify()
+    {
+        $songs = Song::where('is_accepted', '0')->get(['artist', 'title', 'cues', 'is_accepted'])->toArray();
+
+        return $songs;
+    }
+
+    /**
+     * Display song by id
+     * @param $id
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function show($id)
+    {
+        $song = Song::with('user: id,name,avatar')->find($id);
+
+        if (!$song) {
+            return response()->json([
+                'success' => false,
+                'message' => 'Sorry, song with id ' . $id . ' cannot be found.'
+            ], 400);
+        }
+
+        return $song;
+    }
+
+    /**
+     * Display a listing of songs that were uploaded by a specific user.
+     *
+     * @param Song $user_id
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function userSongs($user_id)
+    {
+        $song = Song::where('user_id', $user_id)->first();
+
+        if (!$song) {
+            return response()->json([
+                'success' => false,
+                'message' => 'Sorry, this user did not upload any songs or user does not exist.'
+            ], 400);
+        }
+        else
+        {
+            return $song = Song::where('user_id', $user_id)->get(['id', 'slug', 'artist', 'title'])->toArray();
+        }
+    }
+
+    /**
+     * Store a newly created song.
+     *
+     * @param Request $request
+     * @return \Illuminate\Http\JsonResponse
+     * @throws \Illuminate\Validation\ValidationException
+     */
+    public function store(StoreSong $request)
```

Comment:
1. Nie zgadza się to z założeniami, o których rozmawialiśmy na Slacku. Chcemy mieć osobny route zwracający tekst, żeby można go było wstawić do textarea na froncie. Teraz wygląda to tak, że na sam koniec wprowadzasz do pola `cues`, które jest typu JSON, surowy tekst, bez żadnych znaczników czasu itd. Kompletnie bez sensu.
2. Logika **musi** się znaleźć w osobnym serwisie. Nie możemy tego zostawić w kontrolerze.